src/robot/libdocpkg/model.py

```diff
@@ -97,3 +147,104 @@ def deprecated(self):
     @property
     def _sort_key(self):
         return self.name.lower()
+
+    @property
+    def argument_strings(self):
+        arg_strings = []
+        prev_kind = ''
+        for arg in self.args:
+            if prev_kind != arg.kind:
+                if prev_kind == 'POSITIONAL_ONLY':
+                    arg_strings.append('/')
+                if arg.kind == 'KEYWORD_ONLY' and prev_kind != 'VAR_POSITIONAL':
+                    arg_strings.append('*')
+            arg_strings.append(str(arg))
+            prev_kind = arg.kind
+        if prev_kind == 'POSITIONAL_ONLY':
+            arg_strings.append('/')
+        return arg_strings
+
+
+ARG_TYPES = {'POSITIONAL_ONLY': '',
+             'POSITIONAL_OR_KEYWORD': '',
+             'VAR_POSITIONAL': '*',
+             'KEYWORD_ONLY': '',
+             'VAR_KEYWORD': '**'}
+
+
+class ArgumentDoc(object):
+
+    def __init__(self,
+                 name='',
+                 type=None,
+                 default=None,
+                 kind='POSITIONAL_OR_KEYWORD',
+                 required=True):
+        self.name = name
+        self._type = type
+        self.default = default
+        self.kind = kind
+        self.required = required
+
+    def to_dictionary(self):
+        return {
+            'name': self.name,
+            'type': self.type,
+            'default': self.default.value if self.default else None,
+            'kind': self.kind,
+            'required': self.required
+        }
+
+    @property
+    def type(self):
+        if self._type is None:
+            return
+        if isclass(self._type):
+            return self._type.__name__
+        type_name = str(self._type)
+        if type_name.startswith('typing.'):
+            type_name = type_name[len('typing.'):]
+        return type_name
+
+    def __str__(self):
+        kw_string = ARG_TYPES[self.kind] + self.name
+        kw_string += self._format_type()
+        kw_string += self._format_enum_values()
+        kw_string += self._format_default()
+        return kw_string
+
+    def _format_type(self):
+        if not self.type:
+            return ''
+        return ': {}'.format(self.type)
+
+    def _format_enum_values(self):
+        if isclass(self._type) and issubclass(self._type, Enum):
+            return ' {{ {} }}'.format(self._format_enum(self._type))
+        return ''
+
+    @staticmethod
+    def _format_enum(enum):
+        try:
+            members = list(enum.__members__)
+        except AttributeError:  # old enum module
+            members = [attr for attr in dir(enum) if not attr.startswith('_')]
+        while len(members) > 3 and len(' | '.join(members)) > 42:
+            members[-2:] = ['...']
+        return ' | '.join(members)
+
+    def _format_default(self):
+        if not self.default:
+            return ''
+        default_str = ' = ' if self.type else '='
+        default_str += str(self.default)
+        return default_str
```

Comment:
The top methods have originally been in robotbuilder.py to create the list of strings.
They are now used to convert the ArgumentDoc into a string